<?xml version="1.0"?>
<launch>

	<!-- GDB functionality -->
	<arg name="debug" default="false" />
	<arg unless="$(arg debug)" name="launch_prefix" value="" />
	<arg     if="$(arg debug)" name="launch_prefix" value="gdb --ex run --args" />

	<arg name="hw_or_sim" default="hw" />

	<group ns="frcrobot">

		<!-- Load hardware interface -->
		<node name="frcrobot_hardware_interface" pkg="ros_control_boilerplate" type="frcrobot_$(arg hw_or_sim)_main"
			output="screen" launch-prefix="$(arg launch_prefix)">
		</node>

		<!-- Load controller settings -->
		<rosparam file="$(find ros_control_boilerplate)/config/tune_swerve.yaml" command="load"/>
		<rosparam file="$(find ros_control_boilerplate)/config/robot_code_ready_controller.yaml" command="load"/>

		<!-- Load controller manager -->
		<node name="ros_control_controller_manager" pkg="controller_manager" type="controller_manager" respawn="false"
			output="screen" args="spawn joint_state_controller talon_state_controller fl_angle_controller fr_angle_controller bl_angle_controller br_angle_controller br_drive_controller bl_drive_controller fl_drive_controller fr_drive_controller pdp_state_controller robot_code_ready_controller" />
	</group>

</launch>
